crates/astria-sequencer/src/transaction/unsigned.rs

```diff
@@ -0,0 +1,122 @@
+use astria_proto::sequencer::v1::{
+    unsigned_transaction::Value::AccountsTransaction as ProtoAccountsTransaction,
+    UnsignedTransaction as ProtoUnsignedTransaction,
+};
+use prost::Message as _;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+use crate::{
+    accounts::transaction::Transaction as AccountsTransaction,
+    crypto::SigningKey,
+    hash,
+    transaction::signed::Transaction as SignedTransaction,
+};
+
+/// Represents an unsigned sequencer chain transaction.
+/// This type wraps all the different module-specific transactions.
+/// If a new transaction type is added, it should be added to this enum.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub(crate) enum Transaction {
+    AccountsTransaction(AccountsTransaction),
+}
+
+impl Transaction {
+    /// Attempts to encode the unsigned transaction into bytes.
+    #[must_use]
+    pub(crate) fn to_vec(&self) -> Vec<u8> {
+        match &self {
+            Transaction::AccountsTransaction(tx) => ProtoUnsignedTransaction {
+                value: Some(ProtoAccountsTransaction(tx.to_proto())),
+            },
+        }
+        .encode_length_delimited_to_vec()
+    }
+
+    /// Signs the transaction with the given keypair.
+    #[allow(dead_code)]
+    #[must_use]
+    pub(crate) fn sign(self, secret_key: &SigningKey) -> SignedTransaction {
+        let signature = secret_key.sign(&self.hash());
+        SignedTransaction {
+            transaction: self,
+            signature,
+            public_key: secret_key.verification_key(),
+        }
+    }
+
+    pub(crate) fn hash(&self) -> Vec<u8> {
```

Comment:
Please remove. `util::hash(&my_transaction)` isn't much more work but one less indirection to chase down.

Reply:
you mean to add a helper function that will take in a `Transaction` and hash it? if I remove this right now, I need to do

```
let signature = secret_key.sign(&crate::hash(&self.to_proto().encode_length_delimited_to_vec()));
```

which I don't really like

Reply:
Well, this is crate-private and only used in one location (immediately above in `sign`). So it's not pretty, but at least I would see at a glance what's happening.

Also, f I were to see `self.hash()` I would be very surprised that it's not even the in-memory representation of my type that is being hashed, but the protobuf-serialized rust-protobuf in-memory representation of another type that my type is first converted to.